src/main/java/frc/robot/oi/JoystickResponseCurve.java

```diff
@@ -0,0 +1,145 @@
+package frc.robot.oi;
+
+import edu.wpi.first.math.MathUtil;
+
+/**
+ * Provides different response curves for joystick inputs to improve driver control.
+ */
+public class JoystickResponseCurve {
+    /**
+     * Available joystick response curve types.
+     */
+    public enum CurveType {
+        LINEAR,
+        SQUARE,
+        CUBIC,
+        MILD,
+        EXPONENTIAL,
+        SINUSOIDAL,
+        DUAL_RATE
+    }
+
+    private static final double DEFAULT_DEADBAND = 0.1;
+    private static final double DEFAULT_DUAL_RATE_BREAK = 0.5;
+    private static final double DEFAULT_SCALE = 2.0;
+
+    private final CurveType curveType;
+    private final double deadband;
+    private final double dualRateBreak;
+    private final double scale;
+
+    /**
+     * Creates a joystick response curve with only the curve type specified. Uses default
+     * deadband of 0.1.
+     *
+     * @param curveType The type of response curve to use
+     */
+    public JoystickResponseCurve(CurveType curveType) {
+        this(curveType, DEFAULT_DEADBAND);
+    }
+
+    /**
+     * Creates a joystick response curve with specified curve type and deadband.
+     *
+     * @param curveType The type of response curve to use
+     * @param deadband The deadband to apply to inputs
+     */
+    public JoystickResponseCurve(CurveType curveType, double deadband) {
+        this(curveType, deadband, DEFAULT_DUAL_RATE_BREAK, DEFAULT_SCALE);
+    }
+
+    /**
+     * Creates a joystick response curve with fully custom parameters.
+     *
+     * @param curveType The type of response curve to use
+     * @param deadband The deadband to apply to inputs
+     * @param dualRateBreak The breakpoint for DUAL_RATE curves (0.0-1.0)
+     * @param scale The scale factor for EXPONENTIAL curve
+     */
+    public JoystickResponseCurve(
+            CurveType curveType, double deadband, double dualRateBreak, double scale) {
+        this.curveType = curveType;
+        this.deadband = deadband;
+        this.dualRateBreak = MathUtil.clamp(dualRateBreak, 0.0, 1.0);
+        this.scale = scale;
+    }
+
+    /**
+     * Applies the configured response curve to an input value.
+     *
+     * @param input The raw input value (typically -1.0 to 1.0)
+     * @return The processed output value
+     */
+    public double apply(double input) {
+        double value = MathUtil.applyDeadband(input, deadband);
+        double value_abs = Math.abs(value);
+        return switch (curveType) {
+            // Desmos: f(x) = x
```

Comment:
I'll get a screenshot for these....

Reply:
![Screenshot 2025-03-05 135729](https://github.com/user-attachments/assets/c2ce5906-10ce-4167-91ef-c51f2ee0ebf7)

Reply:
Can't get dual-band to plot - 🤷

Reply:
https://www.desmos.com/calculator/upcjvxwtc4